### PR TITLE
planning: move common function BuildStopDecision()

### PR DESCRIPTION
**Pull request by jmtao**

### CODE REVIEW

File: modules/planning/traffic_rules/BUILD [hunk 1]
```diff
@@ -16,6 +16,7 @@ cc_library(
         "rerouting.cc",
         "stop_sign.cc",
         "traffic_light.cc",
+        "traffic_rule.cc",
```
Comment (xiaoxq) [inline modules/planning/traffic_rules/BUILD]: This package gets larger and larger, better to split.

File: modules/planning/traffic_rules/traffic_light.cc [hunk 2]
```diff
@@ -133,18 +132,27 @@ void TrafficLight::MakeDecisions(
 
     if (signal_color == perception::TrafficLight::GREEN) {
       continue;
+    } else {
```
Comment (xiaoxq) [inline modules/planning/traffic_rules/traffic_light.cc]: No need to use `else`, which reduces indents.

File: modules/planning/traffic_rules/traffic_light.cc [hunk 3]
```diff
@@ -133,18 +132,27 @@ void TrafficLight::MakeDecisions(
...
-    BuildStopDecision(frame, reference_line_info, traffic_light_overlap);
+    std::string virtual_obstacle_id =
+        TRAFFIC_LIGHT_VO_ID_PREFIX + traffic_light_overlap.object_id;
+    const std::vector<std::string> wait_for_obstacles;
+    BuildStopDecision(frame, reference_line_info,
+                      virtual_obstacle_id,
+                      traffic_light_overlap.start_s,
+                      config_.traffic_light().stop_distance(),
+                      StopReasonCode::STOP_REASON_SIGNAL,
+                      wait_for_obstacles);
```
Comment (xiaoxq) [inline modules/planning/traffic_rules/traffic_light.cc]: Simply pass `{}` to construct an empty container.

File: modules/planning/traffic_rules/traffic_rule.h [hunk 4]
```diff
@@ -37,6 +40,15 @@ class TrafficRule {
...
       Frame* const frame, ReferenceLineInfo* const reference_line_info) = 0;
 
+ protected:
+  int BuildStopDecision(Frame* const frame,
+                        ReferenceLineInfo* const reference_line_info,
+                        const std::string& stop_wall_id,
+                        const double stop_line_s,
+                        const double stop_distance,
+                        const StopReasonCode& stop_reason_code,
+                        const std::vector<std::string>& wait_for_obstacles);
```
Comment (xiaoxq) [inline modules/planning/traffic_rules/traffic_rule.h]: Put input arguments before output arguments.

Comment (jmtao) [reply modules/planning/traffic_rules/traffic_rule.h]: fixed.

**Pull request merged**